### DISCUSSION TITLE
新的打包样式在Windows上面没有办法执行

### DISCUSSION THREAD

**Discussion by Create-Peace** [antvis/S2#3052] [General]
pnpm build:esm 在Windows上执行报错，Windows上没有rsync命令，我在Windows上使用roboCopy替代resync，能做个兼容吗？

**Comment by petercat-assistant** [comment]
在 Windows 上执行 `pnpm build:esm` 时遇到的问题是由于缺少 `rsync` 命令。您提到使用 `roboCopy` 作为替代方案，这是一个不错的选择。要实现兼容性，可以在构建脚本中检测操作系统，并根据操作系统选择使用 `rsync` 或 `roboCopy`。

以下是一个简单的示例，说明如何在 Node.js 脚本中实现这种兼容性：

```javascript
const { exec } = require('child_process');
const os = require('os');

const isWindows = os.platform() === 'win32';

const command = isWindows ? 'robocopy /E /COPYALL source destination' : 'rsync -av source/ destination/';

exec(command, (error, stdout, stderr) => {
  if (error) {
    console.error(`执行错误: ${error}`);
    return;
  }
  console.log(`stdout: ${stdout}`);
  console.error(`stderr: ${stderr}`);
});
```

在这个脚本中，我们使用 `os.platform()` 来检测操作系统。如果是 Windows，则使用 `robocopy`，否则使用 `rsync`。请根据您的具体需求调整 `robocopy` 和 `rsync` 的参数。

如果您希望在 antvis/S2 项目中实现此更改，建议您在项目的构建脚本中应用类似的逻辑。

<details>
<summary>🪧 Tips</summary>
For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.
</details>

**Comment by lijinke666** [comment]
可以来个 PR, 我这边没有 Windows 没法调试

**Comment by Alexzjt** [comment]
已解决，https://github.com/antvis/S2/pull/3090
